src/map-app/app/map-ui.ts

```diff
@@ -143,23 +143,18 @@ export class MapUI {
   }
 
   isSelected(initiative: Initiative): boolean {
-    // Currently unimplemented - I can see an sea-initiative-active class
-    // being applied if a test was true, but it makes no sense in the current
-    // context AFAICT. The test was:
-    // initiative === this.contextStack.current()?.initiatives[0]
-
-    // The main thing is seemed to do is highlight an initiative (or
-    // initiatives) in the directory pop-out list of initiatives in a
-    // category.
-    
-    // For now, just return false always. We may re-implement this later.
-    return false; 
+    const selectedInitiatives = this.mapPresenter?.getSelectedInitiatives() ?? [];
+    return  selectedInitiatives.includes(initiative);
   }
 
   toggleSelectInitiative(initiative: Initiative) {
-    // Currently unimplemented.
+    const selectedInitiatives = this.mapPresenter?.getSelectedInitiatives() ?? [];
     
-    // EventBus.Markers.needToShowLatestSelection.pub(lastContent.initiatives);
+    if (selectedInitiatives.includes(initiative)) {
+      EventBus.Markers.needToShowLatestSelection.pub(selectedInitiatives.filter(i => i !== initiative));
+    } else {
+      EventBus.Markers.needToShowLatestSelection.pub([...selectedInitiatives, initiative]);
+    }
   }
```

Comment:
Puzzled by this - seems to be re-implementing initiative selection, something which I had thought maybe used to be a thing, long ago but had been unused and probably broken for ages.

Doesn't seem to be related to counting and displaying visible initiatives?

Reply:
This method wasn't implemented, but I needed it in order to highlight the selected initiative in the sidebar.

The previous code for populating the initiative sidebar tried to achieve this by setting the class `.sea-initiative-active` but this didn't work because the class didn't persist when the sidebar UI was refreshed.

Reply:
I'm uncertain, but I seem to remember that selection stuff is to do with selecting markers on the map, not selecting sidebar items. 

I think here you're just using them to zoom into the marker? Via `EventBus.Markers.needToShowLatestSelection`?

Not a big deal, just checking one sort of functionality isn't being confused for another and revived to serve a different purpose.

Reply:
I think because I'm coming from LX, I'm seeing things in a Redux "app state" kind of way.

We sort of keep a state of the selected markers on the map, within the `currentlySelected` array below. When we click on an initiative in the sidebar, or on a marker directly, it adds the initiative to this array. So I've written this function to expose this state, so that the currently selected initiative is also highlighted in the results pane. It should work however the marker is selected.

**Tangential to this point:** I think this is actually the most frustrating bit of the UI. The app's state is fragmented and stored all over the code, within different classes. And then if another class wants to access or change the state, it's a nightmare to thread the information through a chain of functions, especially if they're in different parts of the UI.
It's a limitation of OOP that I experienced as a Java Android developer, since it often leads to anti-patterns, unless the code is written in a very strict and methodical way with a clear structure (which this isn't).
If we were to re-write some of the UI code, I think this might actually be the most beneficial change- to move app state to a central store such as Redux, which any part of the UI can hook into. Since we implemented Redux more widely in LX, it has made the code a lot cleaner. I think this is more pressing than changing to something like React. The html structure is already fairly understandable and d3 doesn't seem terrible.

Reply:
I think something like this would be a first step to anything similar to introducing react anyway.

However, suspect it still won't be easy! There is a lot of stuff, and untangling all the state might not be simple. Therefore I suspect it would need to be further broken down into steps which can be attempted independently...

I've tried to get rid of the EventBus stuff as much as I could and at least make it typesafe (previously it was easy to send malformed messages to misnamed targets, and what the allowed targets was totally opaque and undocumented).  I've not entirely succeeded but it's better than it was.  What probably needs to be done is to continue with that so that we have a kind if "central source of truth" which behave more like you get in Redux. But there are other bits of state and coupled logic all over the place, and I don't know how deep those depths go...

Which ultimately means we need to get budget-holder clearance to put our snorkels on and try investigating - but leave a route to bail out if it starts getting too tangled or taking too long.

On the other hand - we might want to look at the bigger picture an see if that goes somewhere we need to be, or if we can instead dredge out the useful bits and retrofit them into a new framework - perhaps if we want to be able to make very big maps?   Therefore needs the kind of high-level outcomes planning we scheduled (for today but had to postpone.)